api/ml-jokes/mljokes/data.py

```diff
@@ -0,0 +1,91 @@
+# imports
+import os
+import pickle
+import numpy as np
+import pandas as pd
+from bs4 import BeautifulSoup
+import re
+from nltk import word_tokenize
+from nltk.corpus import stopwords, wordnet as wn
+from nltk.stem.wordnet import WordNetLemmatizer
+
+# Run the first time
+# nltk.download('stopwords')
+# nltk.download('punkt')
+# nltk.download('wordnet')
+
+stopwd = stopwords.words('english')
+
+# Utils
+def extract_joke(path):
+    with open(path) as fp:
+        soup = BeautifulSoup(fp, "html.parser")
+        text = soup.find('table').get_text()
+        text = re.sub('\n', ' ', text)
+        text = re.sub('[\s]+', ' ', text)
+        return text.strip()
+
+def clean_text(text):
+    lemmas = []
+    tokens = word_tokenize(text)
+    lemmas = [WordNetLemmatizer().lemmatize(token).lower() for token in tokens if token.lower() not in stopwd and re.match('^[\w\s]+$', token)]
+    return ' '.join(lemmas)
+
+# Read jokes
+def read_jokes(path_jokes='./data/jokes'):
+    jokes = []
+    jokes_clean = []
+    jokes_len = []
+    joke_ids = []
+
+    for file in os.listdir(path_jokes):
+        text = extract_joke(path_jokes+'/'+file)
+        clean = clean_text(text) 
+        id_ = re.findall('\d+', file)[0]
+        jokes.append(text)
+        jokes_clean.append(clean)
+        jokes_len.append(len(text.split()))
+        joke_ids.append(id_)
```

Comment:
For loops are often a place we find units of work that could be moved to its own function(s). The approach here is common in Object Oriented Programming, but not common in Functional Programming. A common design pattern used to break work like this into more testable units is [Map/Reduce](https://www.learnpython.org/en/Map,_Filter,_Reduce). It looks like you used this approach in `cf_utils::get_all_ratings`. Why not do that here?

(note: I provided a deeper reflection on breaking code into units of work in the comments for topics.py)